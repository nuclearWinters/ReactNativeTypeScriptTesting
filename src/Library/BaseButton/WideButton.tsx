import React from 'react';
import BaseButton, {Props} from './BaseButton';
import {StyleSheet} from 'react-native';

const defaultStyles = StyleSheet.create({
  boxStyle: {
    alignSelf: 'center',
    marginVertical: 64,
    backgroundColor: 'rgb(0,100,180)',
    width: '80%',
    paddingVertical: 12,
    borderRadius: 100,
  },
  textStyle: {
    fontSize: 24,
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

const {boxStyle, textStyle} = defaultStyles;

const WideButton: React.FC<Props> = ({onPress, text, styles}) => (
  <BaseButton
    styles={{
      boxStyle: [boxStyle, styles?.boxStyle],
      textStyle: [textStyle, styles?.textStyle],
    }}
    onPress={onPress}
    text={text}
  />
);
export default WideButton;
