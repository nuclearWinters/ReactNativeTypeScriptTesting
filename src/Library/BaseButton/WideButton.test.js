import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import WideButton from './WideButton';

test('BorderButton', async () => {
  const onPress = () => {};
  const text = 'Test';
  const {getByTestId, getByText, queryByTestId, baseElement} = render(
    <WideButton onPress={onPress} text={text} />,
  );
  const button = getByTestId('box');
  console.log(button.type);
  fireEvent.press(button);
  expect(queryByTestId('box')).toBeTruthy();
  const textButton = getByTestId('text');
  expect(textButton.props.children).toBe(text);
  expect(baseElement).toMatchSnapshot();
});
