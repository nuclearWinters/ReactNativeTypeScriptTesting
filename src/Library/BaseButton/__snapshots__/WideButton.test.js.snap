// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderButton 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      style={
        Array [
          Array [
            Object {
              "alignSelf": "center",
              "backgroundColor": "rgb(0,100,180)",
              "borderRadius": 100,
              "marginVertical": 64,
              "paddingVertical": 12,
              "width": "80%",
            },
            undefined,
          ],
        ]
      }
      testID="box"
    >
      <Text
        style={
          Array [
            Array [
              Object {
                "color": "white",
                "fontSize": 24,
                "fontWeight": "bold",
                "textAlign": "center",
              },
              undefined,
            ],
          ]
        }
        testID="text"
      >
        Test
      </Text>
    </TouchableOpacity>
  </View>
</View>
`;
